filebeat/autodiscover/builder/hints/config_test.go

```diff
@@ -30,7 +30,7 @@ func TestUnpackCopiesDefault(t *testing.T) {
 		"default_config": common.MapStr{
 			"type": "container",
 			"paths": []string{
-				"/var/log/containers/*${data.kubernetes.container.id}.log",
+				"/var/log/containers/*${data.container.id}.log",
```

Comment:
I think we should preserve tests for both since `data.kubernetes.container.id` will be around for a while until we have a solid plan to deprecate it.

Reply:
I'm happy to revert this test but bear in mind that this test doesn't check what is the value of "paths[0]" or if that path contains "data.container.id" or "data.kubernetes.id". Actually https://github.com/elastic/beats/blob/d753ba8dbca6d82698ba180843864676dede3ef5/filebeat/autodiscover/builder/hints/config.go#L33 refer to "data.container.id" already

Reply:
I see, this makes sense. However do we have tests that actually test the resolution of `${data.container.id}` and/or `${data.kubernetes.container.id}`? If not would that make sense to add them so as to ensure this functionality?

Reply:
If I understand correctly https://github.com/elastic/beats/blob/d753ba8dbca6d82698ba180843864676dede3ef5/filebeat/autodiscover/builder/hints/logs_test.go#L38 is creating a default config with `data.container.id` and then https://github.com/elastic/beats/blob/d753ba8dbca6d82698ba180843864676dede3ef5/filebeat/autodiscover/builder/hints/logs_test.go#L181 is checking the evaluation of that variable. We do check for the evaluation of `data.kubernetes.container.name` at https://github.com/elastic/beats/blob/d753ba8dbca6d82698ba180843864676dede3ef5/filebeat/autodiscover/builder/hints/logs_test.go#L887 and https://github.com/elastic/beats/blob/d753ba8dbca6d82698ba180843864676dede3ef5/filebeat/autodiscover/builder/hints/logs_test.go#L907

Reply:
@ChrsMark i have just committed a new test to check for data.kubernetes.container.id as well. As far as I understand though, this test only proves that if the event contains that field, the test will pass. How do we make sure though that the bus.Event contains `data.kubernetes.container.id`. should I check that end to end with a real k8s cluster with autodiscover enabled?